### PR TITLE
feat: add repo specific wallet support

### LINKED ISSUE #719: Enable Support for Repository Specific Wallet
For testing purposes it could be useful to run the production code but on a different (insolvent) wallet. 

I also think that in large organizations it might be better for budgeting purposes to allow for wallet configuration per repository. 

We already have support for this in the organization config so I imagine that adding the capability for repository config should be pretty straightforward. 

_Originally posted by @ubiquibot in https://github.com/ubiquibot/production/issues/50#issuecomment-1704294859_

### PR DESCRIPTION
**Pull request by me505**
Resolves #719

### CODE REVIEW

**Comment by pavlovcik** [comment]
Please demo this working by configuring your test org and showing a repo specific permit and a generic org permit.

**Comment by me505** [comment]
do i have to remove merged configs or no, i dont really get why not if no

**Comment by me505** [comment]
hmm its reading the parsed repo and getting the decrypted private key but its reading undefined after getPrivateKey function call, very weird

**Comment by me505** [comment]
@pavlovcik 
https://github.com/me505/ubiquibot/issues/5

**Comment by me505** [comment]
https://github.com/me505/ubiquibot/blob/development/.github/ubiquibot-config.yml

**Comment by me505** [comment]
the failed attempts were because i was dumb while trying to generate a key

**Review by whilefoo — CHANGES_REQUESTED** [review]
why are you adding typescript interfaces?

**Comment by me505** [comment]
Should I only modify wide repo config?

File: src/utils/private.ts [hunk 1]
```diff
@@ -127,8 +127,15 @@ export const getWideConfig = async (context: Context) => {
     }
   }
   const parsedDefault: MergedConfig = DEFAULT_CONFIG_JSON;
-  const privateKeyDecrypted = parsedOrg && parsedOrg[KEY_NAME] ? await getPrivateKey(parsedOrg[KEY_NAME]) : undefined;
-
+  const privateKeyDecrypted = await (async () => {
```
Comment (whilefoo) [inline src/utils/private.ts:130]: why is this anonymous function?

**Review by whilefoo — CHANGES_REQUESTED** [review]
you can delete org config type because now repo and org configs have the same format

**Comment by me505** [comment]
@whilefoo done

**Comment by me505** [comment]
@whilefoo how to setup org wallet

**Comment by whilefoo** [comment]
> @whilefoo how to setup org wallet

it's `private-key-encrypted` field in org and repo config. you can [generate](https://vanity-eth.tk/) a random one

**Comment by me505** [comment]
https://github.com/me505/ubiquibot/issues/7 for repo wallet

**Comment by me505** [comment]
https://github.com/me505-qa/ubiquibot/issues/1 org permit

**Comment by me505** [comment]
@whilefoo idk how to prove it to you that this is a org permit but this is it

**Review by 0xcodercrane — DISMISSED** [review]
The codebase looks good. will wait for @me505.

**Comment by me505** [comment]
What do you want me for?

**Review by whilefoo — APPROVED** [review]

**Comment by me505** [comment]
@pavlovcik can you review this?

**Review by 0xcodercrane — APPROVED** [review]

**Comment by 0xcodercrane** [comment]
> What do you want me for?

oh sorry my bad. I wanted to tag whilefoo lol

**Comment by pavlovcik** [comment]
Can you guys verify the following scenarios hold true with this implementation?

1. The org sets an org wallet; no repo wallet set; payouts generated from org wallet.
2. The org sets an org wallet; repo wallet set; payouts generated from repo wallet.

**Comment by whilefoo** [comment]
> Can you guys verify the following scenarios hold true with this implementation?
> 
> 1. The org sets an org wallet; no repo wallet set; payouts generated from org wallet.
> 2. The org sets an org wallet; repo wallet set; payouts generated from repo wallet.

I just tested and can confirm

**Pull request merged**